### DISCUSSION TITLE
Uploading multiple image assets into array via API

### DISCUSSION THREAD

**Discussion by BenBao93** [sanity-io/sanity#4083] [Sanity Studio Support]
I am trying to upload multiple image assets into an array field via the javascript SDK via a patch request. The requests succeeds without error and returns the documents content, however without any images added.

This is the code I am using:
```
const imageAssets = []
const imageAsset = await client.assets
    .upload('image', imageData, {
        filename: fileName
    })
imageAssets.push({
    _type: 'image',
    asset: {
        _type: "reference",
        _ref: imageAsset._id
    }
})

try {
    await client
        .patch(postId)
        .insert(
            "replace", "images[0:]", imageAssets
        ).commit()
} catch (e) {
    console.log('error', e.message)
}
```
As far as I can see from the documentation this should work, however there is no trace of any asset in the document and absolutely no error thrown. The call runs through and uploads the images then everything just disappears into a void.

**Comment by jzhao-30** [comment]
Same question, Any solutions？

**Comment by jonathangaldino** [comment]
I would be happy to have such functionality working on sanity API.

The way that I am doing right now is making several API calls, one for each image.
